### PR TITLE
Documentation

### LINKED ISSUE #24: Adding Documentation Folder
adding a documentation folder for each demo that we did and will do 
to share files with all the teammates 
and to have a place to discuss strategy.

### PR DESCRIPTION
**Pull request by MortarDefender**
Demo 1
- Presentation.pdf represent what each team member will say in the first demo, 
   what the site will look like, what are the main features and more
- Requirements.md - has all the requirements that we need to show and do for the first demo
  
Demo 2
- DFD represents our Data Flow Diagram of the site to come
- ERD represents our Entity Relationship Diagram so that we can design our models accordingly
- Requirments.md is the required work that we need to do for the second demo
  
Demo 3 And Demo 4
- In each folder, there is the requirements.md that will show what we will need in that demo.

fix: #24

### CODE REVIEW

**Review by YaelD — DISMISSED** [review]
Looks great

File: documentation/Demo 1/Requirments.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Red Hat Beyond 07
+
+Statement of Work (SOW)
```
Comment (kobihk) [inline documentation/Demo 1/Requirments.md]: maybe you can add a link to the source of this page/content:
https://github.com/redhat-beyond/documents/blob/main/Beyond07-SOW.md

File: documentation/Demo 1/Requirments.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+   issues
+6. Make sure you assign your teammates as well as your mentors as reviewers for
+   your PRs
+7. The review process can take a few days - make sure you send your PRs early
+   enough, so they will be merged in time for the demo
+8. You will be given a template for each demo with the specific details that are
+   required
+9. Designate a primary speaker for your team. Other members are encouraged to
+   speak as well, but if so, plan who talks on what so you can all finish within
+   the allotted time (15m).
```
Comment (kobihk) [inline documentation/Demo 1/Requirments.md]: One option is to add a link to the demo requirements 
with this attitude, the responsibility of the quality of the content is on the maintainer of the SOW
but when you add it to your repository you need to make sure you have a great CR on the content ;-)
```suggestion
   the allocated time (15m).
```

File: documentation/Demo 1/Requirments.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# Red Hat Beyond 07
+
+Statement of Work (SOW)
+***
+
+
+## Milestones
+
+### Demo 1 (March 13th): Project selection and setting up a DEV environment
```
Comment (kobihk) [inline documentation/Demo 1/Requirments.md]: and here link to this part like:
https://github.com/redhat-beyond/documents/blob/main/Beyond07-SOW.md#demo-1-march-13th-project-selection-and-setting-up-a-dev-environment

File: documentation/Demo 2/Requirments.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+Statement of Work (SOW)
+***
+
+
+## Milestones
+
+### Demo 2 (April 3rd): Data model and business logic
+
+The purpose of this demo is to show your application's data model and data
```
Comment (kobihk) [inline documentation/Demo 2/Requirments.md]: one more problem when you copy the content is that you have to make sure all the time nothing is updated on the source page

**Review by kobihk — CHANGES_REQUESTED** [review]
about the DFD file, the "Register" is not lead to the "Main Site" but to the Login page

**Comment by kobihk** [comment]
@MortarDefender please reply to each comment (even if you do not agree... you can explain your thoughts) 
to be able to know you saw all of them.
if you fixed it you can write "Done"

File: documentation/Demo 1/Requirments.md (hunk 1 above)
Comment (MortarDefender) [reply documentation/Demo 1/Requirments.md]: change to link

File: documentation/Demo 1/Requirments.md (hunk 3 above)
Comment (MortarDefender) [reply documentation/Demo 1/Requirments.md]: done

File: documentation/Demo 2/Requirments.md (hunk 4 above)
Comment (MortarDefender) [reply documentation/Demo 2/Requirments.md]: done

**Comment by kobihk** [comment]
@MortarDefender in the ERD file I paid attention we have in many entities the same couple (event_id, user_id) I thought maybe we can add id to participants and use this participant_id instead of the couple (event_id, user_id)
wdyt?

**Comment by MortarDefender** [comment]
@kobihk  something like this?
ERD5
 
if so I think they are a more clean solution but we will need to be more indirect with our data in the terms of searching something in the DB, I have not done it much beforehand but this is what I think. 
I will show the rest of the team and we will decide if to commit the changes or not.

**Comment by kobihk** [comment]
@MortarDefender Yep that's what I meant
but you are right, sometimes in the search/filter it could be a little bit more complicated if you want to search by user_id or event_id.
Check it with the rest of the team and make a decision if is good for you or not.
one more thing maybe it will be more accurate to call the participants > EventParticipant

**Comment by MortarDefender** [comment]
@kobihk We have decided to move forward with your idea

**Review by kobihk — DISMISSED** [review]

**Comment by MortarDefender** [comment]
@kobihk  there seems to be a problem with the flake8 checks, do you know how to fix it?

**Comment by kobihk** [comment]
@MortarDefender 
I got the following answer from @EdDev 
"""
Yesterday there was an outage on the github-actions. You need to re-push (force) with a different hash.
(there is no other way I know of to re-trigger such a problem)
"""
please try it

**Comment by MortarDefender** [comment]
@kobihk done

**Review by kobihk — APPROVED** [review]

**Comment by MortarDefender** [comment]
@kobihk can you merge my request

**Comment by kobihk** [comment]
@MortarDefender Please ask for at least 2 approval from your team members

**Review by danninon — APPROVED** [review]

**Review by YaelD — APPROVED** [review]
Looks good

**Pull request merged**